packages/composite-checkout-wpcom/src/components/wp-contact-form.js

```diff
@@ -14,11 +14,76 @@ import { useHasDomainsInCart } from '../hooks/has-domains';
 import Field from './field';
 import { SummaryLine, SummaryDetails, SummarySpacerLine } from './summary-details';
 import { LeftColumn, RightColumn } from './ie-fallback';
+import { prepareDomainContactDetails } from '../types';
 
-export default function WPContactForm( { summary, isComplete, isActive } ) {
-	const isDomainFieldsVisible = useHasDomainsInCart();
+const contactDetailsFormat = ( { hasDomainsInCart } ) => {
+	if ( hasDomainsInCart ) {
+		return 'DOMAINS';
+	}
+	return 'DEFAULT';
+};
+
+const renderContactDetails = ( {
+	translate,
+	domainName,
+	hasDomainsInCart,
+	contactInfo,
+	renderDomainContactFields,
+	setters,
+} ) => {
+	const format = contactDetailsFormat( { hasDomainsInCart } );
+	const requiresVatId = isEligibleForVat( contactInfo.countryCode.value );
+	switch ( format ) {
+		case 'DOMAINS':
+			return (
+				<React.Fragment>
+					<DomainContactFieldsDescription>
+						{ translate(
+							'Registering a domain name requires valid contact information. Privacy Protection is included for all eligible domains to protect your personal information.'
+						) }
+					</DomainContactFieldsDescription>
+					{ renderDomainContactFields(
+						[ domainName ],
+						prepareDomainContactDetails( contactInfo ),
+						setters.updateContactDetails,
+						setters.applyDomainContactValidationResults
+					) }
+					{ requiresVatId && <VatIdField /> }
+				</React.Fragment>
+			);
+		default:
+			return (
+				<React.Fragment>
+					<TaxFields
```

Comment:
We're still going to need the country drop-down, the phone number masking,  and the postal code validation for `TaxFields`. Is there some way we can leverage the domain contact fields to get these three?

Reply:
I see that you injected calypso's fields for these data in #38869, so I'm not sure how important this concern is now. But I think the short answer is no; I don't see a way to conditionally hide fields in the [ContactDetailsFormFields](https://github.com/Automattic/wp-calypso/tree/master/client/components/domains/contact-details-form-fields), and at the moment we do have less control over these fields for e.g. displaying error states.